crates/services/tx_status_manager/src/service.rs

```diff
@@ -1,71 +1,121 @@
+use crate::{
+    config::Config,
+    manager::TxStatusManager,
+    ports::{
+        P2PPreConfirmationGossipData,
+        P2PPreConfirmationMessage,
+        P2PSubscriptions,
+    },
+    subscriptions::Subscriptions,
+    update_sender::TxStatusChange,
+    TxStatusStream,
+};
 use fuel_core_services::{
     RunnableService,
     RunnableTask,
     ServiceRunner,
     StateWatcher,
     TaskNextAction,
 };
-use fuel_core_types::services::{
-    p2p::{
-        GossipData,
-        PreConfirmationMessage,
-        Sealed,
+use fuel_core_types::{
+    fuel_tx::{
+        Bytes32,
+        TxId,
+    },
+    services::{
+        p2p::{
+            GossipData,
+            PreConfirmationMessage,
+            Sealed,
+        },
+        preconfirmation::Preconfirmation,
+        txpool::TransactionStatus,
     },
-    preconfirmation::Preconfirmation,
-    txpool::TransactionStatus,
 };
 use futures::StreamExt;
+use tokio::sync::{
+    mpsc,
+    oneshot,
+};
 
-use crate::{
-    config::Config,
-    manager::{
-        TimeProvider,
-        TxStatusManager,
+enum ReadRequest {
+    GetStatus {
+        tx_id: TxId,
+        sender: oneshot::Sender<Option<TransactionStatus>>,
     },
-    ports::{
-        P2PPreConfirmationGossipData,
-        P2PPreConfirmationMessage,
-        P2PSubscriptions,
+    Subscribe {
+        tx_id: TxId,
+        sender: oneshot::Sender<anyhow::Result<TxStatusStream>>,
     },
-    subscriptions::Subscriptions,
-    update_sender::TxStatusChange,
-};
+}
 
-#[derive(Clone, Default)]
-pub struct SystemTimeProvider;
-impl SystemTimeProvider {
-    pub fn new() -> Self {
-        Self
+enum WriteRequest {
+    UpdateStatus {
+        tx_id: TxId,
+        status: TransactionStatus,
+    },
+    NotifySkipped {
+        tx_ids_and_reason: Vec<(Bytes32, String)>,
+    },
+}
+
+#[derive(Clone)]
+pub struct SharedData {
+    read_requests_sender: mpsc::Sender<ReadRequest>,
+    write_requests_sender: mpsc::UnboundedSender<WriteRequest>,
```

Comment:
Issue similar to the one I reported [here](https://github.com/FuelLabs/fuel-core/pull/2725#discussion_r1963182638).
Basically, I think the less "unbounded" collections we have in the system the better ;-)

Reply:
We are okay to take that risk, because we rely on the assumption that other services handle back pressure from outside world. We don't want other services to wait when they write to the TxStatusManager